internal/kubernetes/configMap.go

```diff
@@ -80,10 +80,11 @@ func (p *KubernetesProvider) updateConfigMap(cm *corev1.ConfigMap, namespace str
 		return err
 	}
 
-	cm.SetNamespace(namespace)
-	cm.SetResourceVersion(cmOld.ResourceVersion)
-	cm.SetUID(cmOld.UID)
-	_, err = p.foreignClient.Client().CoreV1().ConfigMaps(namespace).Update(cm)
+	cm2 := cm.DeepCopy()
```

Comment:
why did you copied the cm?

Reply:
Because cm is a pointer to a cm stored in memory. In the tests the configmaps are stored in memory, therefore the manipulation of the object pointed by the pointer implies the modification of the original stored object. As result, if I don't copy the whole object, all the modifications of the object are applied to the original element, which is stored in the local cache, leading to an inconsistency between the local and foreign caches. This problem exists only in the testing phase.

Reply:
ok thanks